### ISSUE TITLE
Unable to configure Jade options

### ISSUE THREAD

**Issue by jaydenseric** [keystonejs/keystone#2614]
I need to disable the Jade `pretty` option in both development and production so HTML is minified, to avoid whitespace issues when `display: inline-block` components are looped next to each other.

Looking at the way Keystone renders the views [here](https://github.com/keystonejs/keystone/blob/811e6b3e0914562440f8866feb57fbd5a2211a04/lib/core/render.js#L19), this should at least happen if you set the node env to production. I tried this to no effect.

Regardless, I would like to be able to configure Jade for development.

Is there a workaround in the meantime?

**Comment by mxstbr** [comment]
We're closing all questions and support requests to keep the issue tracker unpolluted. Please ask this question on [Stackoverflow](https://stackoverflow.com/questions/tagged/keystonejs) or [Gitter](https://gitter.im/keystonejs/keystone) instead!

**Comment by jaydenseric** [comment]
This really is an issue. There is no way to configure Jade in Keystone and quality suffers. Having uncompressed HTML in development and compressed production prevents certain layout techniques and causes surprising layout bugs moving between development and production.

```css
.tile {
  display: inline-block;
  width: 50%;
}
```

```html
<div>
  <div class="tile">Foo</div>
  <div class="tile">Foo</div>
</div>
```

Renders the tiles in 2 rows.

```html
<div><div class="tile">Foo</div><div class="tile">Foo</div></div>
```

Displays tiles in 2 columns.

Ideally, Jade alway runs without `--pretty` and you opt in to it if desired.

**Comment by JedWatson** [comment]
Wow, that is genuinely surprising. I was under the impression that whitespace (not spaces between elements but newline+tabs) were ignored by browsers.

We're prepping an 0.3 patch release, I'll see what can be done to open those options up for you.

**Comment by jaydenseric** [comment]
Relevent: https://github.com/pugjs/pug/issues/1833#issuecomment-70699104

**Comment by arthurcamara1** [comment]
+1

**Comment by arthurcamara1** [comment]
This should be configurable: https://github.com/keystonejs/keystone/blob/v0.3.x/lib/core/render.js#L26

**Comment by jaydenseric** [comment]
In the meantime, for others stuck, and not a permanent fix/solution:

```js
response.locals.pretty = false
```

Inside route middleware seems to force Jade to ignore earlier config and render minified HTML.

**Comment by JedWatson** [comment]
@arthurcamara1 that's actually only having an impact on the Admin UI which isn't relevant to this issue; the problem here is being able to configure `pretty` in the default express locals

**Comment by JedWatson** [comment]
I've made it so you can now specify `pretty` in the `locals` setting, and it won't be overridden which fixes this issue! Also published a new patch with the change in it (`0.3.21`) since I didn't get a chance to add this in the last release.
